### ISSUE TITLE
bug: Focus on the wrong window after a confirmation preview window is created

### ISSUE THREAD

**Issue by weizheheng** [stevearc/oil.nvim#208]
### Did you check the docs and existing issues?

- [X] I have read the docs
- [X] I have searched the existing issues

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1361+gd675bd01b-Homebrew

### Operating system/version

MacOS Ventura 13.2.1

### Describe the bug

When there are more than 1 windows opened, for example:

WINDOW 1  |  WINDOW 2

Then perform any actions that will show the oil confirmation preview window like "create" or "delete" files.

After performing the action, closing the oil window by calling `toggle_float()` will always focus the window on the most left but not the previous window (the window the `toggle_float()` is called in the first place)

### Steps To Reproduce

1. nvim -u repro.init
2. Open `window1.txt`
3. Open `window2.txt` in a vertical split (split to the right).
4. Make sure the current focus window is `window2.txt` then call `toggle_float()`
5. Add a new file `window3.txt`
6. Save the changes with `:w`
7. Oil confirmation preview window should show up, then confirm by doing `O`.
8. Close the oil window by calling `toggle_float()` again.
9. Note that, the focus is now on `window1.txt` but not `window2.txt`.

### Expected Behavior

Closing the oil window after performing any actions in the oil confirmation preview window like "create" or "delete" should bring the focus back to window where oil is toggled.

### Directory structure

window1.txt
window2.txt
window3.txt

### Repro

```Lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  {
        "stevearc/oil.nvim",
        config = function()
            require("oil").setup({
              -- add any needed settings here
            })
        end,
  },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

### Did you check the bug with a clean config?

- [X] I have confirmed that the bug reproduces with `nvim -u repro.lua` using the repro.lua file above.

**Comment by weizheheng** [comment]
Thank you so much for this plugin, I am enjoying it!

This might not be a bug in `oil` itself, but I am wondering if anybody else is having the same issue. I have spent some time looking around and found [this](https://github.com/neovim/neovim/issues/20455), which is some what related.

Here is a demo video for the "bug" I described above.


https://github.com/stevearc/oil.nvim/assets/40255418/eeea983f-1fab-4ee5-8f3d-297ab9ca92d0
